File: app/src/main/java/com/cbr/behancesampleapp/domain/network/DataRepository.java
package com.cbr.behancesampleapp.domain.network;


import io.reactivex.Observable;
import io.reactivex.ObservableSource;
import io.reactivex.ObservableTransformer;
import io.reactivex.android.schedulers.AndroidSchedulers;
import io.reactivex.schedulers.Schedulers;

/** Created by dimitrios on 25/08/2017.*/
class DataRepository {

    <T> ObservableTransformer<T, T> applySchedulers() {
        return new ObservableTransformer<T, T>() {
            @Override
            public ObservableSource<T> apply(Observable<T> upstream) {
                return upstream
                        .observeOn(AndroidSchedulers.mainThread())
                        .subscribeOn(Schedulers.io());
            }
        };
    }

}

File: app/src/main/java/com/cbr/behancesampleapp/domain/network/DataRepository.kt
package com.cbr.behancesampleapp.domain.network


import io.reactivex.ObservableTransformer
import io.reactivex.android.schedulers.AndroidSchedulers
import io.reactivex.schedulers.Schedulers

/** Created by dimitrios on 25/08/2017. */
open class DataRepository {
    
    fun <T> applySchedulers(): ObservableTransformer<T, T> {
        return ObservableTransformer { upstream ->
            upstream
                    .observeOn(AndroidSchedulers.mainThread())
                    .subscribeOn(Schedulers.io())
        }
    }
    
}
